test/tests/haproxy-basics/run.sh

```diff
@@ -33,7 +33,7 @@ _request() {
 	fi
 
 	docker run --rm --link "$cid":haproxy "$clientImage" \
-		curl -fs -X"$method" --header 'Host: www.google.com' "$@" "http://haproxy/$url"
+		curl -fsSL -X"$method" --resolve www.google.com:80:$(docker inspect -f '{{.NetworkSettings.IPAddress}}' $cid) "$@" "http://www.google.com/$url"
```

Comment:
Since we're still using `--link "$cid":haproxy`, couldn't this just be `www.google.com:80:haproxy`?

Reply:
Do we have a simple way to ensure this actually hits our `haproxy` instance instead of hitting Google directly inadvertently? (that's about the only thing I like about the previous method -- there's no question at all about whether we're hitting our haproxy instance)

Reply:
> Since we're still using --link "$cid":haproxy, couldn't this just be www.google.com:80:haproxy?

No, curl does not resolve the thing behind the second colon.

Reply:
> Do we have a simple way to ensure this actually hits our haproxy instance instead of hitting Google directly inadvertently?

`curl -v` shows the IP address that is being used, but that is not exactly *simple*.

Example using some local nginx (because the IP address is what matters, not the response itself):
```
[timwolla@~]curl -v --resolve google.com:80:127.0.0.1 http://google.com                                                                                                                                 18:11:09
* Added google.com:80:127.0.0.1 to DNS cache
* Rebuilt URL to: http://google.com/
* Hostname google.com was found in DNS cache
*   Trying 127.0.0.1...
* Connected to google.com (127.0.0.1) port 80 (#0)
> GET / HTTP/1.1
> Host: google.com
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 301 Moved Permanently
< Server: nginx
```

Reply:
That's what I was afraid you'd say. :disappointed:

I think it's important that we find some way to be absolutely sure this resolves properly if we change this, so it still seems better IMO to use the `-H Host: ...` we're doing now since it's more explicit.